src/els_general_provider.erl

```diff
@@ -0,0 +1,125 @@
+              _ -> RootUri0
+            end,
+  InitOptions = case maps:get(<<"initializationOptions">>, Params, #{}) of
+                  null ->
+                    #{};
+                  InitOptions0 ->
+                    InitOptions0
+                end,
+  ok = els_config:initialize(RootUri, Capabilities, InitOptions),
+  DbDir = application:get_env(erlang_ls, db_dir, default_db_dir()),
+  OtpPath = els_config:get(otp_path),
+  els_db:install( node_name(RootUri, els_utils:to_binary(OtpPath))
+                , DbDir
+                ),
+  case maps:get(<<"indexingEnabled">>, InitOptions, true) of
+    true  -> els_indexing:start();
+    false -> lager:info("Skipping Indexing (disabled via InitOptions)")
+  end,
+  {server_capabilities(), State}.
+
+%%==============================================================================
+%% API
+%%==============================================================================
+
+-spec server_capabilities() -> server_capabilities().
+server_capabilities() ->
+  #{ capabilities =>
+       #{ textDocumentSync =>
+            #{ openClose => true
+             , change    => ?TEXT_DOCUMENT_SYNC_KIND_FULL
+             , save      => #{includeText => true}
+             }
+        , hoverProvider =>
+            els_hover_provider:is_enabled()
+        , completionProvider =>
+            #{ resolveProvider => false
+             , triggerCharacters => [<<":">>, <<"#">>, <<"?">>, <<".">>]
+             }
+        , definitionProvider =>
+            els_definition_provider:is_enabled()
+        , referencesProvider =>
+            els_references_provider:is_enabled()
+        , documentHighlightProvider =>
+            els_document_highlight_provider:is_enabled()
+        , documentSymbolProvider =>
+            els_document_symbol_provider:is_enabled()
+        , workspaceSymbolProvider =>
+            els_workspace_symbol_provider:is_enabled()
+        , codeActionProvider =>
+            els_code_action_provider:is_enabled()
+        , documentFormattingProvider =>
+            els_formatting_provider:is_enabled_document()
+        , documentRangeFormattingProvider =>
+            els_formatting_provider:is_enabled_range()
+        , foldingRangeProvider =>
+            els_folding_range_provider:is_enabled()
+        , implementationProvider =>
+            els_implementation_provider:is_enabled()
+        , executeCommandProvider =>
+            els_execute_command_provider:options()
+        , codeLensProvider =>
+            els_code_lens_provider:options()
+        }
+   }.
+
+%%==============================================================================
+%% Internal Functions
+%%==============================================================================
+-spec node_name(uri(), binary()) -> atom().
+node_name(RootUri, OtpPath) ->
+  <<SHA:160/integer>> = crypto:hash(sha, <<RootUri/binary, OtpPath/binary>>),
+  list_to_atom(lists:flatten(io_lib:format("erlang_ls_~40.16.0b", [SHA]))).
```

Comment:
unrelated observation.  It would be useful for debugging if this was made up with an additional component being the last part of the root directory, as reported in the info command.

Reply:
Yes, I'm also not happy about the current naming convention. Feel free to open an issue about this and we tackle this.